typedb/concept/type/thing_type.py

```diff
@@ -18,103 +18,192 @@
 
-    def as_thing_type(self) -> "RemoteThingType":
-        return self
+    def unset_abstract(self, transaction: _Transaction) -> None:
+        thing_type_unset_abstract(transaction.concepts.native_transaction(), self.native_object)
+
+    def set_plays(self, transaction: _Transaction,
+                  role_type: _RoleType, overridden_role_type: Optional[_RoleType] = None) -> None:
+        thing_type_set_plays(transaction.concepts.native_transaction(),
+                             self.native_object, role_type.native_object,
+                             overridden_role_type.native_object if overridden_role_type else None)
+
+    def unset_plays(self, transaction: _Transaction, role_type: _RoleType) -> None:
+        thing_type_unset_plays(transaction.concepts.native_transaction(), self.native_object, role_type.native_object)
+
+    def set_owns(self, transaction: _Transaction, attribute_type: _AttributeType,
+                 overridden_type: Optional[_AttributeType] = None, annotations: Optional[set[Annotation]] = None) -> None:
+        overridden_type_native = overridden_type.native_object if overridden_type else None
+        annotations_array = [anno.native_object for anno in annotations] if annotations else []
+        thing_type_set_owns(
+            self.native_transaction(transaction),
+            self.native_object,
+            attribute_type.native_object,
+            overridden_type_native,
+            annotations_array,
+        )
+
+    def get_owns(self, transaction: _Transaction, value_type: Optional[ValueType] = None,
+                 transitivity: Transitivity = Transitivity.Transitive, annotations: Optional[set[Annotation]] = None
+                 ) -> Iterator[AttributeType]:
+        from typedb.concept.type import attribute_type
+
+        return (attribute_type._AttributeType(item) for item in
+                Streamer(thing_type_get_owns(
+                    self.native_transaction(transaction),
+                    self.native_object,
+                    value_type.native_object if value_type else None,
+                    transitivity.value,
+                    [anno.native_object for anno in annotations] if annotations else []
+                ), concept_iterator_next))
+
+    def get_owns_explicit(self, transaction: _Transaction, value_type: Optional[ValueType] = None,
+                          annotations: Optional[set[Annotation]] = None):
+        return self.get_owns(transaction, value_type, Transitivity.Explicit, annotations)
+
+    def get_plays(self, transaction: _Transaction, transitivity: Transitivity = Transitivity.Transitive) -> Iterator[_RoleType]:
+        from typedb.concept.type import role_type
+
+        return (role_type._RoleType(item) for item in
+                Streamer(thing_type_get_plays(
+                    self.native_transaction(transaction),
+                    self.native_object,
+                    transitivity.value
+                ), concept_iterator_next)
```

Comment:
Is there a difference between this and `map(role_type._RoleType, Streamer(...))`? The `map()` version seems a litlle tighter than the generator.